### ISSUE TITLE
window.onbeforeunload

### ISSUE THREAD

**Issue by TinyMCE-User** [tinymce/tinymce#2274]
Description of problem:
window.onbeforeunload handler is overridden when calling to tinymce.
Means: if I define window.onbeforeunload handler function and then tinymce, the handler is not working.

Steps to reproduce:

// WITH TINYMCE: the handler is not running when page is about to be closed
&ltg;form>
&ltg;textarea name="content" id="mceEditor">&ltg;/textarea>
&ltg;/form>
&ltg;script>
window.onbeforeunload = function() { return 'You are about to close the page; }
&ltg;/script>

// WITHOUT TINYMCE: the handler running fine when page is about to be closed
&ltg;form>
&ltg;textarea name="content">&ltg;/textarea>
&ltg;/form>
&ltg;script>
window.onbeforeunload = function() { return 'You are about to close the page; }
&ltg;/script>

Expected result:
First example should also work, and handler should run as expected.

Actual result:
The handler is not working.

> _**Legacy information imported from TinyMCE bug tracker:**_
>
> _[#T7562](http://archive.tinymce.com/develop/bugtracker_view.php?id=7562) posted by [dcohend](http://archive.tinymce.com/profile.php?name=dcohend)_

> _**Tags:** [firefox msie safari chrome onbeforeunload]_
> _**Status:** Open_
> _**Resolution:** None_
> _**Attached URL:** none_

**Comment by blackcathikari** [comment]
Hi, we've investigated this issue and cannot reproduce it in TinyMCE v4.4.0. Thanks, the TinyMCE Team.

**Comment by Jogai** [comment]
Example here: https://jsfiddle.net/2nudb0pb/
If the handler is deleted (comment line 16 to try) the autosave function starts working

**Comment by sztrzask** [comment]
Can we reopen this? @blackcathikari you were obviously either wrong or lying as this was never properly solved. I've created PR to fix this as it's also affecting me.

**Comment by Jogai** [comment]
Maybe reword your comment a bit

> Never attribute to malice that which can be adequately explained by ignorance

**Comment by blackcathikari** [comment]
Apologies. I'm currently on holidays but I'll ask someone to look into it.
